plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -142,33 +158,34 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRe
      */
     @Override
     public CompletionStage<GetDataObjectResponse> getDataObjectAsync(GetDataObjectRequest request, Executor executor) {
-        final String tenantId = request.tenantId() != null ? request.tenantId() : DEFAULT_TENANT;
-        final GetItemRequest getItemRequest = GetItemRequest
-            .builder()
-            .tableName(getTableName(request.index()))
-            .key(
-                Map
-                    .ofEntries(
-                        Map.entry(HASH_KEY, AttributeValue.builder().s(tenantId).build()),
-                        Map.entry(RANGE_KEY, AttributeValue.builder().s(request.id()).build())
-                        // TODO need to fetch SEQ_NO_KEY
-                    )
-            )
-            .build();
+        final GetItemRequest getItemRequest = getGetItemRequest(request.tenantId(), request.id(), request.index());
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<GetDataObjectResponse>) () -> {
             try {
                 final GetItemResponse getItemResponse = dynamoDbClient.getItem(getItemRequest);
                 ObjectNode sourceObject;
                 boolean found;
+                String sequenceNumberString = null;
                 if (getItemResponse == null || getItemResponse.item() == null || getItemResponse.item().isEmpty()) {
                     found = false;
                     sourceObject = null;
                 } else {
                     found = true;
-                    sourceObject = JsonTransformer.convertDDBAttributeValueMapToObjectNode(getItemResponse.item());
+                    sourceObject = JsonTransformer.convertDDBAttributeValueMapToObjectNode(getItemResponse.item().get(SOURCE).m());
+                    if (getItemResponse.item().containsKey(SEQ_NO_KEY)) {
+                        sequenceNumberString = getItemResponse.item().get(SEQ_NO_KEY).n();
+                    }
```

Comment:
Shouldn't there always be a `SEQ_NO_KEY` if the object exists?  We should put 0 on initial indexing.

Reply:
Yes, `SEQ_NO_KEY` will be set always be set when a new DDB entry is created. However, having contains check for code completeness.